Index: alsa-lib-1.0.22/src/dlmisc.c
===================================================================
--- alsa-lib-1.0.22.orig/src/dlmisc.c	2009-12-16 16:18:51.000000000 +0100
+++ alsa-lib-1.0.22/src/dlmisc.c	2010-01-17 16:33:04.000000000 +0100
@@ -26,7 +26,7 @@
  *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
  *
  */
-
+#define _GNU_SOURCE
 #include "list.h"
 #include "local.h"
 
Index: alsa-lib-1.0.22/src/timer/timer_hw.c
===================================================================
--- alsa-lib-1.0.22.orig/src/timer/timer_hw.c	2010-01-17 16:31:13.000000000 +0100
+++ alsa-lib-1.0.22/src/timer/timer_hw.c	2010-01-17 16:33:04.000000000 +0100
@@ -23,6 +23,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#define __USE_GNU
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include "timer_local.h"
